transport/http/outbound.go

```diff
@@ -121,11 +131,39 @@ func (t *Transport) NewOutbound(chooser peer.Chooser, opts ...OutboundOption) *O
 	for _, opt := range opts {
 		opt(o)
 	}
-	o.client = t.client
-	o.sender = &transportSender{Client: t.client}
+
+	client := t.client
+	if o.tlsConfig != nil {
+		client = createTLSClient(o)
+		o.urlTemplate.Scheme = "https"
+	}
+	o.client = client
+	o.sender = &transportSender{Client: client}
 	return o
 }
 
+func createTLSClient(o *Outbound) *http.Client {
+	transport, ok := o.transport.client.Transport.(*http.Transport)
+	if !ok {
+		// This should not happen as default yarpc http.Client uses
+		// http.Transport and it's not configurable by the user.
+		panic("failed to create http tls client, provided http.Client does not use http.Transport")
```

Comment:
do you want to print the type of `o.transport.client.Transports` in the panic? 
`panic(fmt.Sprintf("failed to create http tls client, provided HTTP client %T does not implement http.Transport"))`

Reply:
done